src/WindowsConsole.php

```diff
@@ -0,0 +1,163 @@
+                SHORT Y;
+            } COORD;
+            
+            // https://learn.microsoft.com/en-us/windows/console/key-event-record-str
+            typedef struct _KEY_EVENT_RECORD {
+                int bKeyDown;
+                WORD wRepeatCount;
+                WORD wVirtualKeyCode;
+                WORD wVirtualScanCode;
+                union {
+                    CHAR AsciiChar;
+                    WORD UnicodeChar;
+                } uChar;
+                DWORD dwControlKeyState;
+            } KEY_EVENT_RECORD;
+            
+            // https://learn.microsoft.com/en-us/windows/console/mouse-event-record-str
+            typedef struct _MOUSE_EVENT_RECORD {
+                COORD dwMousePosition;
+                DWORD dwButtonState;
+                DWORD dwControlKeyState;
+                DWORD dwEventFlags;
+            } MOUSE_EVENT_RECORD;
+            
+            // https://learn.microsoft.com/en-us/windows/console/focus-event-record-str
+            typedef struct _FOCUS_EVENT_RECORD {
+                BOOL bSetFocus;
+            } FOCUS_EVENT_RECORD;
+            
+            // https://learn.microsoft.com/en-us/windows/console/input-record-str
+            typedef struct _INPUT_RECORD {
+                WORD EventType;
+                union {
+                    KEY_EVENT_RECORD KeyEvent;
+                    MOUSE_EVENT_RECORD MouseEvent;
+                    FOCUS_EVENT_RECORD FocusEvent;
+                } Event;
+            } INPUT_RECORD;
+            
+            // https://learn.microsoft.com/en-us/windows/console/getstdhandle
+            HANDLE GetStdHandle(DWORD nStdHandle);
+            // https://learn.microsoft.com/en-us/windows/console/getconsolemode
+            BOOL GetConsoleMode(HANDLE hConsoleHandle, DWORD* lpMode);
+            // https://learn.microsoft.com/en-us/windows/console/setconsolemode
+            BOOL SetConsoleMode(HANDLE hConsoleHandle, DWORD dwMode);
+            // https://learn.microsoft.com/en-us/windows/console/readconsoleinput
+            // ReadConsoleInputW (Unicode) and ReadConsoleInputA (ANSI)
+            BOOL ReadConsoleInputW(HANDLE hConsoleInput, INPUT_RECORD* lpBuffer, DWORD nLength, DWORD* lpNumberOfEventsRead);
+            BOOL ReadConsoleInputA(HANDLE hConsoleInput, INPUT_RECORD* lpBuffer, DWORD nLength, DWORD* lpNumberOfEventsRead);
+            CLang;
+
+        $this->ffi = FFI::cdef($header, 'kernel32.dll');
+
+        // Use the class constants to get the handle
+        /**
+         * @phpstan-ignore-next-line */
+        $this->handle = $this->ffi->GetStdHandle(self::STD_INPUT_HANDLE);
+
+        if (FFI::isNull($this->handle)) {
+            throw new RuntimeException('Failed to get console handle');
+        }
+    }
+
+    /**
+     * @return void
+     */
+    public function __destruct()
+    {
+        /**
+         * @phpstan-ignore-next-line */
+        $this->ffi->SetConsoleMode($this->handle, $this->originalSettings);
```

Comment:
is there a particular reason to do this here vs. `RawMode::disable()` ?

Reply:
you're right, I was originally doing this because I was testing things. I'll remove these deconstrctures